telegram-bot-api/Client.cpp

```diff
@@ -7704,6 +8050,232 @@ td::Status Client::process_get_memory_stats_query(PromisedQueryPtr &query) {
+             });
+
+  return Status::OK();
+}
+
+td::Status Client::process_join_chat_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_id = query->arg("chat_id");
+  auto invite_link = query->arg("invite_link");
+
+  if (!chat_id.empty()) {
+    check_chat(chat_id, AccessRights::Read, std::move(query), [this](int64 chat_id, PromisedQueryPtr query) {
+      send_request(make_object<td_api::joinChat>(chat_id),
+                   std::make_unique<TdOnJoinChatIdCallback>(this, std::move(query), chat_id));
+    });
+  } else if (!invite_link.empty()) {
+    send_request(make_object<td_api::joinChatByInviteLink>(invite_link.str()),
+                 std::make_unique<TdOnReturnChatCallback>(this, std::move(query)));
+  } else {
+    fail_query(400, "Bad request: Please specify chat_id or invite_link", std::move(query));
+  }
+
+  return Status::OK();
+}
+
+td::Status Client::process_add_chat_member_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_id = query->arg("chat_id");
+  TRY_RESULT(user_id, get_user_id(query.get()))
+
+  check_chat(chat_id, AccessRights::Write, std::move(query),
+             [this, user_id](int64 chat_id, PromisedQueryPtr query) mutable {
+               auto chat = get_chat(chat_id);
+               if (chat->type == ChatInfo::Type::Supergroup) {
+                 std::vector<td::int32> user_ids{user_id};
+                 send_request(make_object<td_api::addChatMembers>(chat_id, std::move(user_ids)),
+                              std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+               } else if (chat->type == ChatInfo::Type::Group) {
+                 send_request(make_object<td_api::addChatMember>(chat_id, user_id, 0),
+                              std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+               }
+             });
+  return Status::OK();
+}
+
+td::Status Client::process_report_chat_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_id = query->arg("chat_id");
+  TRY_RESULT(reason, get_report_reason(query.get()));
+  TRY_RESULT(message_ids, get_int_array_arg<td::int64>(query.get(), "message_ids", true));
+
+  check_chat(chat_id, AccessRights::Read, std::move(query),
+             [this, reason = std::move(reason), message_ids = std::move(message_ids)](int64 chat_id,
+                                                                                      PromisedQueryPtr query) mutable {
+               send_request(make_object<td_api::reportChat>(chat_id, std::move(reason), std::move(message_ids)),
+                            std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+             });
+  return Status::OK();
+}
+
+td::Status Client::process_create_chat_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_type = query->arg("type");
+  auto title = query->arg("title");
+
+  if (chat_type == "supergroup") {
+    send_request(make_object<td_api::createNewSupergroupChat>(title.str(), false, "", nullptr),
```

Comment:
Add an optional description parameter

Reply:
I decided not to have that parameter, because it can only be set for supergroups or channels. Basic Groups can't have a description. I wanted to keep them as similar as possible and a description can still be set in a separate method call. Having 2 ways to set a description for a group is bad api design IMO

Reply:
So one would have to make 2 api calls to create a channel with a description? I'd just have it as an optional parameter, we could just ignore it for normal group chats and assume empty if not given

Reply:
Okay, I will add that.